hashtagsv2/hashtags/helpers.py

```diff
@@ -20,9 +20,23 @@ def hashtag_queryset(request_dict):
 
     hashtag_list = split_hashtags(request_dict['query'])
 
-    queryset = Hashtag.objects.filter(
-        hashtag__in=hashtag_list
+    if 'search_type' in request_dict:
+        if request_dict['search_type'] == 'and':
+            rcids_for_hashtag=[]
+            for hashtag in hashtag_list:
```

Comment:
This gets a little hard to follow - could you add some comments explaining the steps involved? :)